examples/on-failure/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/joho/godotenv"
+
+	"github.com/hatchet-dev/hatchet/pkg/client"
+	"github.com/hatchet-dev/hatchet/pkg/cmdutils"
+	"github.com/hatchet-dev/hatchet/pkg/worker"
+)
+
+type stepOneOutput struct {
+	Message string `json:"message"`
+}
+
+func StepOne(ctx worker.HatchetContext) (result *stepOneOutput, err error) {
+	return nil, fmt.Errorf("test on failure")
+}
+
+func OnFailure(ctx worker.HatchetContext) (result *stepOneOutput, err error) {
+	return &stepOneOutput{
+		Message: "Failure!",
+	}, nil
+}
+
+func main() {
+	err := godotenv.Load()
+
+	if err != nil {
+		panic(err)
+	}
+
+	c, err := client.New()
+
+	if err != nil {
+		panic(err)
+	}
+
+	w, err := worker.NewWorker(
+		worker.WithClient(
+			c,
+		),
+	)
+
+	if err != nil {
+		panic(err)
+	}
+
+	err = w.On(
+		worker.NoTrigger(),
+		&worker.WorkflowJob{
+			Name:        "on-failure-workflow",
+			Description: "This runs at a scheduled time.",
+			Steps: []*worker.WorkflowStep{
+				worker.Fn(StepOne).SetName("step-one"),
+			},
+			OnFailure: &worker.WorkflowJob{
```

Comment:
should we abstract this so its just a step in the sdks...? i feel like it might be confusing to introduce a new "jobs" concept out of no where

Reply:
This isn't out of nowhere, it's always been on the Go SDK. It's just that `worker.Fn` implements `WorkflowJob` as well.